packages/openai/src/openai.ts

```diff
@@ -996,13 +1093,20 @@ export class OpenAIProvider<TConfig extends OpenAIConfig = OpenAIConfig> impleme
       } catch (e: any) {
         if (isContextWindowError(e)) {
           const details = parseContextWindowError(e);
-          return {
+          const errorResponse: Response = {
             content: '',
             toolCalls: [],
-            finishReason: 'length',
+            finishReason: 'length' as FinishReason,
             model,
             usage: { text: { input: details?.contextWindow } },
           };
+          
+          // Call post-request hook even on error
+          if (effectiveConfig.hooks?.chat?.postRequest) {
+            await effectiveConfig.hooks.chat.postRequest(request, params, errorResponse, ctx, metadata);
```

Comment:
Let's simplify these if and call statement to be a 1 liner:

await effectiveConfig.hooks?.chat?. postRequest?.(request, params, error response, ctx, metadata)